### PR TITLE
add new settings to disable editing `Main menus` and `Flat menus` in the Wagtail CMS

### PR DESCRIPTION
**Pull request by mvdwaeter**
#234

### CODE REVIEW

**Comment by ababic** [comment]
Hi @mvdwaeter,

Thanks for this! I'm happy with the implementation, and the docs additions are perfect too. It would be nice to have a couple of simple regression tests to show the effect of setting each to `False`. Is that something you're happy to do?

**Comment by mvdwaeter** [comment]
Hi @ababic , I added tests. As Django settings are always a pain to test, I ended up using a mock library  + reloading the wagtail_hooks file, to proof that the `modeladmin_register` function with respective menuAdmin type is called or not based on the setting.

**Comment by ababic** [comment]
Thanks @mvdwaeter! Wow, that was fast! However, I feel these tests could be simplified a little. How do you feel about just checking if accessing listing page returns a 404?

Also, there's an existing [`TestSuperUser`](https://github.com/rkhleics/wagtailmenus/blob/master/wagtailmenus/tests/test_backend.py#L154) test case in [`test_backend.py`](https://github.com/rkhleics/wagtailmenus/blob/master/wagtailmenus/tests/test_backend.py) that would make a suitable home.

**Comment by mvdwaeter** [comment]
Hi @ababic , I started with that first, exactly with your provided test as a template. But unfortunately a 200 response was still returned because the modeladmin was already registered after settings were overridden. Perhaps I overlooked something, but I couldn't get it to work that way. 

The tests I provided are actually very pure, it tests whether the modeladmin_register is called or not based on settings. Assuming that modeladmin_register is well tested by wagtail, that's all you need.

**Comment by ababic** [comment]
HI @mvdwaeter 

> But unfortunately a 200 response was still returned because the modeladmin was already registered after settings were overridden. Perhaps I overlooked something, but I couldn't get it to work that way.

Ah yes, that's probably the case. Not to worry then.

> Assuming that modeladmin_register is well tested by wagtail, that's all you need.

I agree with this to a point. I just don't think it's good practice to have tests assume too much about the technical implementation if at all avoidable - especially so if they're reliant on the inner workings of other apps. It sounds as though this might be the only option though.

File: wagtailmenus/tests/test_disabling_menus_in_cms_settings.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
```
Comment (ababic) [inline wagtailmenus/tests/test_disabling_menus_in_cms_settings.py]: For organisation's sake, I also feel that `test_backend.py` would be a better home for these tests (even if they can't be simplified)

File: wagtailmenus/tests/test_disabling_menus_in_cms_settings.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+from wagtailmenus import wagtail_hooks
+
+try:
+    from importlib import reload  # Python 3.4+
+except ImportError:
+    try:
+        from imp import reload  # Python 3
+    except ImportError:
+        import reload  # Python 2.x
```
Comment (ababic) [inline wagtailmenus/tests/test_disabling_menus_in_cms_settings.py]: Wagtailmenus is Python 3 only now, so it should be possible to simplify the imports a little

Comment (mvdwaeter) [reply wagtailmenus/tests/test_disabling_menus_in_cms_settings.py]: In tox I only see Python 3.4 and up, should Python3 versions lower than 3.4 also be supported?

Comment (ababic) [reply wagtailmenus/tests/test_disabling_menus_in_cms_settings.py]: Just 3.4 and up will do :)

File: wagtailmenus/tests/test_backend.py [hunk 3]
```diff
@@ -1,23 +1,32 @@
+from __future__ import absolute_import, unicode_literals
```
Comment (ababic) [inline wagtailmenus/tests/test_backend.py]: Can we drop this line as well please? This is only really needed to handle unicode consistently in Python 2, which we don't need to worry about (it still lingers in a few files, but I'll eradicate it in time!) :)

**Comment by ababic** [comment]
Thanks for this @mvdwaeter.  Really appreciate your time and effort with this - especially with how tricky the tests have been. Just one small change, and we're good to go! Also, please credit yourself in `CONTRIBUTORS.md` :)

**Comment by mvdwaeter** [comment]
Thank you :) Happy to contribute!

**Pull request merged**